packages/lodestar/src/sync/backfill/backfill.ts

```diff
@@ -0,0 +1,220 @@
+    const {root, epoch} = this.chain.getFinalizedCheckpoint();
+    if (epoch <= 0) {
+      this.logger.debug("BackfillSync no necessary on genesis");
+      return 0;
+    }
+
+    this.logger.debug("BackfillSync initializing from cp", {epoch, root: toHexString(root)});
+
+    // Try to figure out if BackfillSync is necessary. Note that this check is very inexact
+    // - Define "target range" to backfill as: [oldestSlotRequired, cpStartSlot]
+    // - If there's at least 1 block in target range, consider yourself BackfillSynced
+    const cpStartSlot = computeStartSlotAtEpoch(epoch);
+    const oldestSlotRequired = this.getOldestRequiredSlot();
+    const slots = await this.db.blockArchive.keys({gte: oldestSlotRequired, lte: cpStartSlot, limit: 1});
+    if (slots.length > 0) {
+      this.logger.debug("BackfillSync has at least one block in range, skipping", {oldestSlotRequired, cpStartSlot});
+      return slots[0];
+    }
+
+    // Attempt to find the anchor block in the DB
+    try {
+      this.anchorBlock = await this.db.blockArchive.getByRoot(root);
+    } catch (e) {
+      this.logger.error("Error fetching blockArchive", {root: toHexString(root)}, e);
+    }
+
+    // Start processor loop
+    this.processor.trigger();
+
+    for await (const _ of this.processor) {
+      const peer = shuffleOne(this.peers.values());
+      if (!peer) {
+        this.logger.debug("BackfillSync no peers yet");
+        continue;
+      }
+
+      try {
+        // First get anchor block
+        if (!this.anchorBlock) {
+          const anchorBlockRoot = this.chain.getFinalizedCheckpoint().root;
+          await this.syncBlockByRoot(peer, anchorBlockRoot);
+          if (this.anchorBlock) {
+            this.logger.debug("BackfillSync fetched anchorBlock", {root: toHexString(anchorBlockRoot)});
+          } else {
+            // If syncBlockByRoot could not find anchorBlock, loop and try again with another peer
+            continue;
+          }
+        }
+
+        const oldestSlotRequired = this.getOldestRequiredSlot();
+        const anchorBlockSlot = this.anchorBlock.message.slot;
+        if (anchorBlockSlot <= oldestSlotRequired) {
+          // Sync completed, return the oldestSlotSynced
+          return anchorBlockSlot;
+        }
+
+        const toSlot = anchorBlockSlot;
+        const fromSlot = Math.max(toSlot - this.opts.batchSize, oldestSlotRequired);
+        this.logger.debug("BackfillSync syncing range", {fromSlot, toSlot});
+        await this.syncRange(peer, fromSlot, toSlot, this.anchorBlock.message.parentRoot);
+      } catch (e) {
+        this.logger.debug("BackfillSync sync error", e);
+
+        if (e instanceof BackfillSyncError) {
+          switch (e.type.code) {
+            case BackfillSyncErrorCode.INVALID_SIGNATURE:
+            case BackfillSyncErrorCode.NOT_LINEAR:
+            case BackfillSyncErrorCode.NOT_ANCHORED:
+              this.network.reportPeer(peer, PeerAction.LowToleranceError, "BadSyncBlocks");
```

Comment:
Report peers only when they do something bad according to internal rules of the BackfilSync. Other reqresp errors like bad ssz formating, timeouts etc are handled and reported in the ReqResp class